webapp-django/crashstats/crashstats/utils.py

```diff
@@ -254,41 +254,35 @@ def build_default_context(product=None, versions=None):
     a dict. If there's any left-over, raise a 404 error
     """
     context = {}
-    api = models.CurrentProducts()
-    context['currentproducts'] = api.get()
-
-    currentversions = [
-        v for p in context['currentproducts']['products']
-        for v in context['currentproducts']['hits'][p]
-    ]
-    context['currentversions'] = currentversions
+    api = models.ProductVersions()
+    active_versions = OrderedDict()  # so that products are in order
+    # Turn the list of all product versions into a dict, one per product.
+    for pv in api.get(active=True)['hits']:
+        if pv['product'] not in active_versions:
+            active_versions[pv['product']] = []
+        active_versions[pv['product']].append(pv)
+    context['active_versions'] = active_versions
 
     if versions is None:
         versions = []
-    else:
+    elif isinstance(versions, basestring):
         versions = versions.split(';')
 
-    for release in context['currentversions']:
-        if product == release['product']:
-            context['product'] = product
-            if release['version'] in versions:
-                versions.remove(release['version'])
-                if 'versions' not in context:
-                    context['versions'] = []
-                context['versions'].append(release['version'])
-
-    if product is None:
-        # thus a view that doesn't have a product in the URL
-        # e.g. like /query
-        if not context.get('product'):
-            context['product'] = settings.DEFAULT_PRODUCT
-    elif product != context.get('product'):
-        raise http.Http404("Not a recognized product")
-
-    if product and versions:
-        raise http.Http404("Not a recognized version for that product")
-
-    context['releases'] = build_releases(context['currentversions'])
```

Comment:
You did not answer this. Did you investigate if it could be removed? As far as I can tell nothing is using that function anymore.